.github/configs/schema-validator/.spectral.yaml

```diff
@@ -1,11 +1,89 @@
+            validateItemTypeAgainstElementType]
 rules:
-  major-version-in-path: warn
-  openapi-tags: warn
-  oas3-api-servers: warn
-  oas3-unused-component: warn
+  enum-case-convention: off
+  major-version-in-path: off
+  openapi-tags: off
+  oas3-api-servers: off
+  oas3-unused-component: off
+  property-case-convention: off
+  property-description: off
+  property-inconsistent-name-and-type: off
+  
+  element-valid-item-type:
+    message: 'Item type does not exist: {{error}}'
+    given: $.components.schemas[*].allOf[1][x-ob-item-type]
+    severity: error
+    then:
+      function: validateXOBFields
+      functionOptions:
+        objectPath: x-ob-item-types
+  
+  element-valid-item-type-group:
+    message: 'Item type group does not exist: {{error}}'
+    given: $.components.schemas[*].allOf[1][x-ob-item-type-group]
+    severity: error
+    then:
+      function: validateXOBFields
+      functionOptions:
+        objectPath: x-ob-item-type-groups
 
+  item-type-group-valid-type:
+    message: 'Item type group type does not exist: {{error}}'
+    given: $[x-ob-item-type-groups][*].type
+    severity: error
+    then:
+      function: refPathExists
+  
+  item-type-group-valid-entries:
+    description: The enums or units in the item type group exist in the item type definition.
+    message: '{{error}}'
+    given: $[x-ob-item-type-groups][*]~
+    severity: error
+    then:
+      function: validateItemTypeGroupEntries
+  
+  element-valid-sample-value:
+    description: >-
+      Checks the following items about sample value:
+      - It is an object. The object may be empty (see requireAtLeastOneField).
+      - Its fields are only primitives.
+      - Value has the correct OpenAPI type
+      - If the schema definition OB item type defines enums or units, and it has an item type group:
+        - When enums are defined, Value is an enum of the OB item type and item type group.
+        - When units are defined, Unit is a unit of the OB item type and item type group.
+      - Additional validation rules for each primitve can be added to validateXOBSampleValue.
+    message: '{{error}}'
+    # Give entire 'allOf' array of an element definition
+    # Spectral resolves all $ref's, so allOf[0].properties.Value exists
+    given: $.components.schemas[*][?(@property === 'allOf' && @[0].properties.Value)]
+    severity: error
+    then:
+      function: validateXOBSampleValue
+      functionOptions:
+        # Require sample value be a nonempty object
+        requireAtLeastOneField: false
+        enumItemTypeIgnoreList:
+          - UUIDItemType # Value is checked against the regex of UUID
+
+  element-valid-openapi-type-and-item-type:
+    description: >-
+      Check that the element's OpenAPI type and OB item type are compatible.
+      OpenAPI 'integer' and 'number' types are only compatible with OB unit item types.
+      OpenAPI 'boolean' and 'string' types are only compatible with OB enum item types.
```

Comment:
All OB enums are strings, but I don't think OpenAPI boolean are strings: they are literals true and false.